crates/config/src/sections/upstream_oauth2.rs

```diff
@@ -383,8 +383,8 @@ fn signed_response_alg_default() -> JsonWebSignatureAlg {
 
 #[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
 pub struct SignInWithApple {
-    /// The private key used to sign the `id_token`
-    pub private_key: String,
+    /// The private key file used to sign the `id_token`
+    pub private_key_file: String,
```

Comment:
This is a breaking change I'd like to avoid introducing. Could you do instead allow both properties? This is what we do for some secrets:

https://github.com/element-hq/matrix-authentication-service/blob/a3202a690ad677dacbed264c877e5e6831986012/crates/config/src/sections/secrets.rs#L34-L39

Then in the configuration section validation function, we check that only one of them is specified:

https://github.com/element-hq/matrix-authentication-service/blob/a3202a690ad677dacbed264c877e5e6831986012/crates/config/src/sections/secrets.rs#L145-L155